src/init.sh

```diff
@@ -170,5 +180,173 @@ function init_help()
     '  init - Creates a kworkflow.config file in the current directory.' \
     '  init --arch <arch> - Set the arch field in the kworkflow.config file.' \
     '  init --remote <user>@<ip>:<port> - Set remote fields in the kworkflow.config file.' \
-    '  init --target <target> Set the default_deploy_target field in the kworkflow.config file'
+    '  init --target <target> - Set the default_deploy_target field in the kworkflow.config file' \
+    '  init --interactive - Perform first time setup in an interactive manner. Recommended for new users.'
+}
+
+# This function gets four configuration values from git: user name, user email,
+# editor, and initial branch name and stores them inside an associative array
```

Comment:
> Why do we need the branch name?

We selected to configure user name, email, editor, and branch name since these are the configurations in this tutorial: https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup. We think that they put the branch name as an important setup since there is a movement to change the default branch name from master to main, but, for now, master is still used as default. So, this is an important setup to do.

Reply:
Ok, but in that case, we should check the current branch and use it as a default. For example, in the case of amd driver the default branch is named `amd-staging-drm-next`, and right after clone, this is the main branch. In that case, we should suppose the current branch is the main branch but give the user the chance to change it.